### PR TITLE
Fix archival feature

### LINKED ISSUE #52: History and Visibility Archival statuses show up incorrectly
Namespace's HistoryArchivalStatus and VisibilityArchivalStatus show incorrect data

Repro:
1. Enable History Archival
2. Navigate to Namespace Settings

Expected:
"History Archival" and "Visibility Archival" show Enabled

Actual:
Disabled

### PR DESCRIPTION
**Pull request by feedmeapples**
1. Fixes #52 History and Visibility archival features status now correctly show up as Enabled (when enabled)
2. Archival page is now visible
2. Fixes the Workflow History status filtering
3. Fixes Query filter

image

### CODE REVIEW

File: client/routes/namespace/workflow-archival/constants.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-  { value: '3', label: 'Terminated' },
-  { value: '4', label: 'Continued As New' },
-  { value: '5', label: 'Timed Out' },
+  { value: 'CLOSED', label: 'Closed' },
+  { value: 'COMPLETED', label: 'Completed' },
+  { value: 'FAILED', label: 'Failed' },
+  { value: 'CANCELED', label: 'Cancelled' },
+  { value: 'TERMINATED', label: 'Terminated' },
+  { value: 'CONTINUED_AS_NEW', label: 'Continued As New' },
+  { value: 'TIMED_OUT', label: 'Timed Out' },
```
Comment (alexshtin) [inline client/routes/namespace/workflow-archival/constants.js:8]: These looks like `WorkflowExecutionStatus`. Where do they come from? Also there is no `CLOSED` but there are `RUNNING`. Are you sure they must be UPPERCASE?

File: client/helpers/map-namespace-description.js [hunk 2]
```diff
@@ -21,9 +21,9 @@ export default function(namespace) {
       : 'Unknown',
     'Emit Metrics': emitMetric ? 'Yes' : 'No',
     'History Archival':
-      historyArchivalStatus === 'ENABLED' ? 'Enabled' : 'Disabled',
+      historyArchivalStatus == 'Enabled' ? 'Enabled' : 'Disabled',
```
Comment (alexshtin) [inline client/helpers/map-namespace-description.js:24]: What is the point of this at all? Why not:
```suggestion
      historyArchivalStatus,
```

File: server/middleware/workflow-client.js [hunk 3]
```diff
@@ -129,6 +129,7 @@ function enumTransform(item) {
     'continue_as_new_initiator',
     'event_type_timer',
     'timeout_type',
+    'archival_status',
```
Comment (alexshtin) [inline server/middleware/workflow-client.js:132]: Can you add of them here? At least `retry_status`.

File: client/routes/namespace/workflow-archival/constants.js (hunk 1 above)
Comment (feedmeapples) [reply client/routes/namespace/workflow-archival/constants.js:8]: > Where do they come from?

This is the execution status. These constants are used to show the options of the status filter. 
image
Similar thing happens on worklow executions page

> Are you sure they must be UPPERCASE?

The casing doesn't matter as the server always turns them lowercase. https://github.com/temporalio/temporal/blob/master/common/archiver/filestore/queryParser.go#L250

> Also there is no CLOSED but there are RUNNING
My understanding is that archived workflow history is not going to have RUNNING status. QueryParser that uses ExecutionStatus doesn't expect RUNNING status neither. As for the CLOSED status - it was a case before as you can see in the old code. I'm going to remove it completely

File: server/middleware/workflow-client.js (hunk 3 above)
Comment (feedmeapples) [reply server/middleware/workflow-client.js:132]: I've added retry_status as part of current PR. I will add the rest in the near time in a separate PR

**Pull request merged**